pkg/util/error_list.go

```diff
@@ -45,10 +45,67 @@ func (list errorList) Error() string {
 	return result
 }
 
+// Except returns any errors (or nested errors, if the list contains
+// nested []error slices) for which all fns return false. If no errors
+// remain a nil list is returned. The resulting list will have all
+// nested []error slices flattened as a side effect.
```

Comment:
flattening sub-slices seems wrong to me - They are each logically a single error.